lib/jobs/node-discovered-alert-job.js

```diff
@@ -43,16 +45,22 @@ function nodeAlertJobFactory(
      */
     NodeAlertJob.prototype._run = function run() {
         var self = this;
+        var nodeIp = lookupService.nodeIdToIpAddresses(self.nodeId);
+        var nodeMac = lookupService.ipAddressToMacAddress(nodeIp);
+        var nodeInfo = {
+                          'ipAddress': nodeIp,
+                          'macAddress': nodeMac,
+ 			   'nodeId': self.nodeId
+                       };
+
         return waterline.nodes.needByIdentifier(self.nodeId)
         .then(function(node) {
             assert.string(node.type);
 
             if(self.context.data){
-                return eventsProtocol.publishNodeEvent(node, 'discovered', self.context.data);
-            }
-            else{
-                return eventsProtocol.publishNodeEvent(node, 'discovered');
+                nodeInfo.contextData = self.context.data;
```

Comment:
I'm not sure the context.data details, if it's proper to put mac and ip to context.data to decrease nested values.